int8 state
int8 locatingGate        = 0
int8 aligningWithGate    = 1
int8 passingGate         = 2
int8 searchingForPath    = 3
int8 locatingDie         = 4
int8 touchingDie         = 5

